tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/wwwroot/app/dashboard/dashboard.controller.js

```diff
@@ -19,222 +31,304 @@
+    	            { field: 'documentName', displayName: 'Document', width: '25%', cellTemplate: '<div class="ui-grid-cell-contents">{{row.entity.documentName}}</div>', enableColumnMenu: false },
+                    { field: 'documentClientId', displayName: 'Client', width: '15%', cellTemplate: '<div class="ui-grid-cell-contents" >{{row.entity.documentClientId}}</div>', enableColumnMenu: false },
+                    { field: 'documentOwner', displayName: 'Author', width: '14%', enableColumnMenu: false },
+                    { field: 'documentModifiedDate', displayName: 'Modified date', width: '20%', enableColumnMenu: false },
+                    { field: 'documentId', displayName: 'Document ID', width: '10%', enableColumnMenu: false },
+                    { field: 'documentVersion', displayName: 'Version', width: '6%', enableColumnMenu: false },
+                    { field: 'pin', width: '5%', cellTemplate: '<div class="ui-grid-cell-contents"><img src="../Images/pin-666.png"/></div>', enableColumnMenu: false }
+                ],
+                onRegisterApi: function (gridApi) {
+                    vm.gridApi = gridApi;
                     //Set the selected row of the grid to selectedRow property of the controller
-    		        gridApi.selection.on.rowSelectionChanged($scope, function (row) {
-    		            vm.selectedRow = row.entity
-    		        });
-    		    }
-    	    }
-    	   
+                    gridApi.selection.on.rowSelectionChanged($scope, function (row) {
+                        vm.selectedRow = row.entity
+                    });
+                }
+            }
+
             //function to toggle check all 
-    	    vm.toggleChecker = function (checked) {
-    	        var rows = vm.gridApi.core.getVisibleRows(vm.gridApi.grid),
+            vm.toggleChecker = function (checked) {
+                var rows = vm.gridApi.core.getVisibleRows(vm.gridApi.grid),
                     allChecked = true;
 
-    	        for (var r = 0; r < rows.length; r++) {
-    	            if (rows[r].entity.checker !== true) {
-    	                allChecked = false;
-    	                break;
-    	            }
-    	        }    	        
-    	        $("#chkAllDocCheckBox").prop('checked', allChecked);
-    	    };
+                for (var r = 0; r < rows.length; r++) {
+                    if (rows[r].entity.checker !== true) {
+                        allChecked = false;
+                        break;
+                    }
+                }
+                $("#chkAllDocCheckBox").prop('checked', allChecked);

```

Comment:
bind checked property using model in angular way instead of finding element and applying property. This helps to keep code clean and improves UI performance.